File: Lesson6 Simple-Intermediate Math/Problem2/Solutions/Solution.cpp
#include <iostream>
#include<algorithm>
using namespace std;

int main()
{
   int n; int x; int cnt = 0; 
   cin>>n; cin>>x;
   int array[101];
   for(int i = 0; i < n; i++){
       cin>>array[i];
   }
   for(int i = 0; i < n; i++){
       for(int j = i + 1; j < n; j++){
           for(int k = j + 1; k < n; k++){
               int temp[3]; 
               temp[0] = array[i];
               temp[1] = array[j];
               temp[2] = array[k];
               sort(temp, temp + 3);
               if(temp[1] == x){
                   cnt++;
               }
           }
       }
   }
   cout<<cnt<<endl;
   return 0;
}

File: Lesson6 Simple-Intermediate Math/Problem2/Solutions/Solution.py
# python 2 code
s = map(int, raw_input().split())

n = s[0]
x = s[1]

t = map(int, raw_input().split())

array = []
cnt = 0

for num in t:
    array.append(num)

for i in range (n):
    for j in range(i + 1, n):
        for k in range(j + 1, n):
            temp = []
            temp.append(array[i])
            temp.append(array[j])
            temp.append(array[k])
            temp.sort()
            if temp[1] == x:
                cnt +=1
print cnt
